### ISSUE TITLE
Design and implementation of performance and footprint testing tools for the specific case of AIX `wazuh-manager` local mode

### ISSUE THREAD

**Issue by jmv74211** [wazuh/wazuh-qa#3680]
| Wazuh QA branch | 
| ---- |
| `3680-footprint-testing-tool` | 

### Description

In order to perform the necessary performance tests proposed in [wazuh-qa#3679](https://github.com/wazuh/wazuh-qa/issues/3679), it is necessary to design and implement a series of tools that allow us to automatically perform the following tasks:

- [x] Manipulate `n` files every `x` time interval for a time `t`.
- [x] Generate `n` log lines of `x` (variable) size in `y` different files simultaneously every `z` time interval for a total time `t`.
- [x] Simulate the syslog server to which the `wazuh-manager` will send alerts. Necessary to count and measure the received alerts.
- [x] Measure the number of alerts received during each time `t`, and store these values in a data file (`csv`).
- [x] Measure CPU, memory, disk usage ... during the process
- [x] Graph the results obtained, both footprint and number of alerts received.
- [x] Main script that uses each of the other tools in threads and that allows us to synchronize and automate the main process.

> Note: Manipulate = create, update and delete

**Comment by Rebits** [comment]
### Update 15/12/2022
- Created syslog server tool to count and measure the received alerts.
In order to test the tool, it has been used the `syslog_simulator` tool. Using a custom script we have generated 30,80,90 and 200 events in 2 seconds intervals:
```
python3 client.py  -n 30 -m "Testing" -a 127.0.0.1 -p 519 --protocol udp
sleep 2;
python3 client.py  -n 80 -m "Testing" -a 127.0.0.1 -p 519 --protocol udp
sleep 2;
python3 client.py  -n 90 -m "Testing" -a 127.0.0.1 -p 519 --protocol udp
sleep 2;
python3 client.py  -n 200 -m "Testing" -a 127.0.0.1 -p 519 --protocol udp
sleep 2;
``` 

Expected events were gathered by the Syslog Server:

```
root@ubuntu:/home/vagrant# python3 syslog.py  -t 8 -f testing.log -p udp -P 519  -i 2
2022-12-15 16:07:14,665 — INFO — Starting syslog server
2022-12-15 16:07:22,665 — INFO — Shutting down syslog server
INFO:syslog_server_simulator:Shutting down syslog server
[30, 80, 90, 200]
2022-12-15 16:07:23,167 — INFO — Total messages 400
INFO:syslog_server_simulator:Total messages 400
root@ubuntu:/home/vagrant# 
```

**Comment by jmv74211** [comment]
## Update 20/12/2022

Added a [script](https://github.com/wazuh/wazuh-qa/commit/520559bcbf080c52fa01622c9373f831b19d3de3) to generate performance charts per monitored daemon and stat.

**Comment by Rebits** [comment]
### Update 20/12/2022
- Finish the main script 
- Refactor file stress script
- Implement CSV data handling
- First testing with a manager
- Meeting with @jmv74211

**Comment by Rebits** [comment]
### Regarding UDP packets lost in localhost manager testing

It has developed all the tools for footprint tests in the AIX system. However, `syslog_output` only supports UDP protocol. This condition makes it impossible to avoid packets lost. In addition, the amount of average UDP packets lost will depend on multiple factors OS, network drivers, kernel settings among others.
The only valid approach is to measure in the environment how many UDP packets are lost in average using already created tools

**Comment by Rebits** [comment]
### Update 22/12/2022
- Meeting with @jmv74211 
- Increase logging
- Refactor tools
- PoC in a manager environment

**Comment by Rebits** [comment]
### Update  23/12/2022

- Fix minor errors detected for 3.1.0 version
- Fix errors in graph creation
- Improve data visualization
